act/retrievals/sonde.py

```diff
@@ -477,3 +422,260 @@ def calculate_pbl_liu_liang(
+    ds,
+    temperature='tdry',
+    pressure='pres',
+    height='alt',
+    smooth_height=3,
+    base=5,
+):
+    """
+    Function for calculating the PBL height from a radiosonde profile
+    using the Heffter technique. There are differences from the ARM
+    VAP at times due to different averaging schemes.  Larger differences
+    do occur at times and are unknown as to the cause but it is being
+    investigated and is potential a code issue with the VAP.
+
+    Parameters
+    ----------
+    ds : xarray Dataset
+        Dataset housing radiosonde profile for calculations
+    temperature : str
+        The name of the temperature field.
+    pressure : str
+        The name of the pressure field.
+    height : str
+        The name of the height field
+    smooth_height : int
+        Number of points to do a moving average on sounding height data to reduce noise
+    base : int
+        Interval for pressure gridding.  In testing, 5 mb was found to produce results with
+        the lowest RMS
+
+    Returns
+    -------
+    ds : xarray Dataset
+        xarray dataset with results stored in pblht_liu_liang variable
+
+    References
+    ----------
+    Heffter JL. 1980. “Transport Layer Depth Calculations.” Second Joint Conference on
+        Applications of Air Pollution Meteorology, New Orleans, Louisiana.
+
+    Sivaraman, C., S. McFarlane, E. Chapman, M. Jensen, T. Toto, S. Liu, and M. Fischer.
+        "Planetary boundary layer (PBL) height value added product (VAP): Radiosonde retrievals."
+        Department of Energy Office of Science Atmospheric Radiation Measurement (ARM) Program
+        (United States) (2013).
+
+    """
+
+    # Preprocess the sonde data to ensure the same methods across all retrievals
+    ds2 = preprocess_sonde_data(ds, temperature=temperature, pressure=pressure,
+                                height=height, smooth_height=smooth_height, base=base)
+
+    # Get data
+    pres = ds2[pressure].values
+    alt = ds2[height].values
+    theta = ds2['potential_temperature'].values
+
+    # Calculate the lapse rate
+    theta_gradient = np.diff(theta) / np.diff(alt)
+
+    # Calculate AGL
+    if np.isnan(alt[0]):
+        idx = np.where(~np.isnan(alt))[0]
+        agl = alt - alt[idx[0]]
+    else:
+        agl = alt - alt[0]
+
+    # Find where the lapse rate is greater than 0.005 K/m
+    idx = np.where(theta_gradient >= 0.005)[0]
+
+    # Find the consistent layers by grouping the indices together
+    # Does not include a single height as a layer
+    ranges = []
+    for key, group in groupby(enumerate(idx), lambda i: i[0] - i[1]):
+        group = list(map(itemgetter(1), group))
+        if group[-1] - group[0] > 0:
+            ranges.append((group[0], group[-1]))
+
+    # Subset ranges to lowest 5
+    if len(ranges) > 5:
+        ranges = ranges[0:5]
+
+    # For each layer, calculate the difference in theta from
+    # top and bottom of the layer.  The lowest layer where the
```

Comment:
Multiple whitespace before the the, this is just more nitpicking though.

Reply:
I'm a double spacer between sentences! :)

Reply:
Ah then it's fine! I didn't know if that was intentional or accident! @AdamTheisen can resolve this then.